### PR TITLE
Use ByteCode::new_raw_checked and propagate error

### LINKED ISSUE #10172: validate EOF bytecode on Genesis.
### Describe the feature

If there ar EOF bytecodes in the Genesis file we should validate those bytes so that only valid EOF bytecodes are found in the state.

The function is found in `use revm::interpreter::analysis::validate_raw_eof_inner` and it needs to be `CodeType::ReturnOrStop` type (Will renaming it to `ContainerKind::Runtime`)

### Additional context

It does not affect mainnet, it is a guard for new chains that have EOF enabled from the start.

### PR DESCRIPTION
**Pull request by danielcdz**
Closes #10172

### CODE REVIEW

File: crates/storage/db-common/src/init.rs
Comment (mattsse) [inline crates/storage/db-common/src/init.rs:171]: can we add an error trace for the account (address) here, this would make debugging easier

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/storage/db-common/src/init.rs
Comment (emnul) [reply crates/storage/db-common/src/init.rs:171]: In the original issue, @rakita mentioned 
> the error needs to be propagated as invalid json.

Do you know what this means?

Comment (emnul) [reply crates/storage/db-common/src/init.rs:171]: @danielcdz you can add an error trace with the [error!](https://docs.rs/tracing/latest/tracing/macro.error.html#examples) macro

Also you will need to transform the [EofDecodeError](https://github.com/emnul/reth/blob/4124144049a8c87c25d5ffb74718307df70aa512/crates/storage/errors/src/provider.rs#L19-L20) you get from the Bytecode::new_raw_checked function into a [ProviderError](https://github.com/emnul/reth/blob/4124144049a8c87c25d5ffb74718307df70aa512/crates/storage/errors/src/provider.rs#L19-L20) to get the code to compile. It looks like the `EofDecodeError` enum would need to be added as a member to the `ProviderError` enum, but this may lead to a breaking change and you would need to add the `revm-primitives` dependency to the `crates/storage/errors` crate so I would defer to a maintainer on the best approach.

**Review by mattsse — APPROVED** [review]

**Review by emnul — COMMENTED** [review]
LGTM

**Comment by danielcdz** [comment]
@emnul thank you for all the help and knowledge shared here!

**Pull request merged**
